File: app/src/main/java/com/blackmirror/dongda/utils/OtherUtils.java
package com.blackmirror.dongda.utils;

import android.content.Context;
import android.net.Uri;

/**
 * Created by Ruge on 2018-04-04 下午12:19
 */
public class OtherUtils {
    private static final String ANDROID_RESOURCE = "android.resource://";
    private static final String FOREWARD_SLASH = "/";

    public static Uri resourceIdToUri(Context context, int resourceId) {
        return Uri.parse(ANDROID_RESOURCE + context.getPackageName() + FOREWARD_SLASH + resourceId);
    }


}

File: app/src/main/java/com/blackmirror/dongda/utils/OtherUtils.kt
package com.blackmirror.dongda.utils

import android.content.Context
import android.net.Uri

/**
 * Create By Ruge at 2018-12-07
 */
private val ANDROID_RESOURCE = "android.resource://"
private val FOREWARD_SLASH = "/"

fun resourceIdToUri(context: Context, resourceId: Int): Uri {
    return Uri.parse(ANDROID_RESOURCE + context.packageName + FOREWARD_SLASH + resourceId)
}